### PR TITLE
Fix intermittent test failures due to results ordering issues.

### PR DESCRIPTION
**Pull request by frankban**
Make bundles-containing and charm-related API
endpoints return results in a specific order.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/172/

File: internal/v4/archive.go [hunk 1]
```diff
@@ -324,6 +325,7 @@ func verificationError(err error) error {
 	for i, err := range verr.Errors {
 		messages[i] = err.Error()
 	}
+	sort.Strings(messages)
```
Comment (rogpeppe) [inline internal/v4/archive.go:329]: It occurs to me that now we've got multiError, we could probably use it here.
We're probably still need to the sorting to get consistent keys though.

This could probably do with a doc comment anyway.

File: internal/v4/relations.go [hunk 2]
```diff
@@ -187,18 +189,19 @@ func (h *handler) metaBundlesContaining(entity *mongodoc.Entity, id *charm.Refer
 	}
 
 	// Prepare and return the response.
-	response := make([]*params.MetaAnyResponse, 0, len(entities))
```
Comment (rogpeppe) [inline internal/v4/relations.go:190]: Why the change from pointer to value?

File: internal/v4/relations.go [hunk 3]
```diff
@@ -134,6 +135,7 @@ func (h *handler) getRelatedIfaceResponses(
 			}
 		}
 	}
+	sort.Sort(byId(responses))
```
Comment (rogpeppe) [inline internal/v4/relations.go]: I'm not sure we need to sort the responses if the entities are already sorted,
which could probably be done by mongodb server side. Do we mind
that much about numeric revision ordering?

File: internal/v4/archive.go (hunk 1 above)
Comment (frankban) [inline internal/v4/archive.go:329]: Done.

File: internal/v4/relations.go (hunk 3 above)
Comment (frankban) [inline internal/v4/relations.go]: Ok, fixed.

File: internal/v4/relations.go (hunk 2 above)
Comment (frankban) [inline internal/v4/relations.go:190]: Done.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/173/

**Comment by urosj** [comment]
:+1:

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge
